tutorials/plugins/gdnative/gdnative-cpp-scons.rst

```diff
@@ -0,0 +1,872 @@
+
+.. code-block:: python
+
+    env = Environment()
+
+Not so fast though! When Windows compilation is done using Visual Studio, it will require environment variables that Windows has already setup to access Visual Studio's compiler ``cl``. As such, we can ensure that scons finds the appropriate compiler for all operating systems by by cloning the existing environment in the Windows case.
+
+To do this, we'll need to import the ``os`` Python module (to get the OS's environment). We'll also need to determine when the user is compiling on Windows versus other operating systems.
+
+.. code-block:: python
+    import os
+
+    platform = ARGUMENTS.get("platform", "windows")
+
+    env = Environment()
+    if platform == "windows":
+        env = Environment(ENV = os.environ)
+
+The ``ARGUMENTS`` global is an object that allows us to ``get`` a named parameter from the command line as well as supply a default value if the parameter is not given. In this case, we are asking the user to provide us with the ``platform`` parameter. If they do not provide one, we will assume it is "windows" since that is the most common case.
+
+If we are compiling for Windows, then we set our ``env`` to be a new ``Environment`` in which its internal environment is a copy of the operating system's environment variables.
+
+OS-Specific Information
+-----------------------
+
+We'll probably want to give our resulting library a unique name depending on what platform and bit version it's aimed at (32-bit vs. 64-bit). With that in mind, let's also collect the desired bits version.
+
+.. code-block:: python
+
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+    lib_name = ARGUMENTS.get("name", "libdefault")
+
+    library_file = lib_name + "." + platform + "." + str(bits)
+
+We'll also want to allow the user to decide where to output their various dynamic libraries (one for each platform/bit), so let's also collect the library path.
+
+If we combine the library filename with the platform/bits-specific directory path, then we can determine the full appropriate filename to use for the dynamic library's output location.
+
+.. code-block:: python
+
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+    lib_name = ARGUMENTS.get("name", "libdefault")
+    lib_path = ARGUMENTS.get("lib", "bin/")
+
+    # Force expected formatting.
+    # Ensures we view Windows-style and filepath-like 
+    # directory paths as typical UNIX directory paths.
+    lib_path = lib_path.split("\\").join("/")
+    if lib_path[-1:] != "/":
+        lib_path += "/"
+
+    platform_dir = ""
+
+    if platform == "osx":
+        platform_dir = "osx" + str(bits) + "/"
+    elif platform == "linux":
+        platform_dir = "x11" + str(bits) + "/"
+    elif platform == "windows":
+        platform_dir = "win" + str(bits) + "/"
+
+    library_file = lib_name + "." + platform + "." + str(bits)
```

Comment:
Add a comment mentioning how scons will automatically insert the OS-specific library file extension when building. No need to include logic for it.